src/icon.c

```diff
@@ -141,8 +171,19 @@ GdkPixbuf *get_pixbuf_from_file(const char *filename)
 {
         char *path = string_to_path(g_strdup(filename));
         GError *error = NULL;
+        gint w, h;
 
-        GdkPixbuf *pixbuf = gdk_pixbuf_new_from_file(path, &error);
+        if (!gdk_pixbuf_get_file_info (path, &w, &h)) {
+                LOG_W("Failed to load image info for %s", filename);
+                g_error_free(error);
+                return NULL;
```

Comment:
The last leak from valgrind is actually ours, because of the early return here `path` isn't freed.

Reply:
Should be fixed now